test/unit/fact_value_test.rb

```diff
@@ -68,5 +68,23 @@ def setup
     results = FactValue.search_for("host = #{host.fqdn}")
     assert_empty results
   end
-end
 
+  describe '.my_facts' do
+    let(:target_host) { FactoryGirl.create(:host, :with_hostgroup, :with_facts) }
+    let(:other_host) { FactoryGirl.create(:host, :with_hostgroup, :with_facts) }
+
+    test 'returns all facts for admin' do
+      as_admin do
+        assert_empty (target_host.fact_values + other_host.fact_values).map(&:id) - FactValue.my_facts.map(&:id)
+      end
+    end
+
+    test 'returns visible facts for filtered user' do
+      user_role = FactoryGirl.create(:user_user_role)
+      FactoryGirl.create(:filter, role: user_role.role, permissions: Permission.where(name: 'view_hosts'), search: "hostgroup_id = #{target_host.hostgroup_id}")
```

Comment:
Nitpick - hashrocket syntax seems to be the standard, I'm all for changing it, we can easily enforce it with Rubocop.